quickwit/quickwit-common/src/checklist.rs

```diff
@@ -84,12 +88,47 @@ pub fn print_checklist(check_list_results: &[(&str, anyhow::Result<()>)]) {
 /// Run a checklist and print out its successes and failures on stdout.
 ///
 /// If an error is encountered, the proccess will exit with exit code 1.
-pub fn run_checklist(checks: Vec<(&str, anyhow::Result<()>)>) {
+pub fn run_checklist(checks: Vec<(&str, anyhow::Result<()>)>) -> Result<(), ChecklistError> {
     print_checklist(&checks);
     if !checks
         .iter()
         .all(|(_, check_items_res)| check_items_res.is_ok())
     {
-        std::process::exit(1);
+        return Err(ChecklistError::from(checks));
+    }
+
+    Ok(())
+}
+
+#[derive(Error, Debug)]
+pub struct ChecklistError {
+    pub errors: Vec<(String, anyhow::Result<()>)>,
+}
+
+impl From<Vec<(&str, anyhow::Result<()>)>> for ChecklistError {
```

Comment:
I don't think we need that `From`. Let's use a proper constructor in `ChecklistError`. Maybe `ChecklistError::from_errors`?